src/Cassandra/Connections/Connection.cs

```diff
@@ -732,11 +765,12 @@ public OperationState Send(IRequest request, Action<Exception, Response> callbac
 
             IncrementInFlight();
 
-            var state = new OperationState(callback)
-            {
-                Request = request,
-                TimeoutMillis = timeoutMillis
-            };
+            var state = new OperationState(
+                callback,
+                request,
+                timeoutMillis > 0 ? timeoutMillis : Configuration.SocketOptions.ReadTimeoutMillis,
```

Comment:
This is not needed (it was changed on `master` so this "decision" is made before this point)